src/dmd/dsymbolsem.d

```diff
@@ -2831,6 +2831,67 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
         if (!sc)
             return;
 
+        bool repopulateMembers = false;
+        if (ns.identExp)
+        {
+            // resolve the namespace identifier
+            sc = sc.startCTFE();
+            Expression resolved = ns.identExp.expressionSemantic(sc);
+            resolved = resolved.expressionSemantic(sc);
+            resolved = resolveProperties(sc, resolved);
+            sc = sc.endCTFE();
+            resolved = resolved.ctfeInterpret();
+            StringExp name = resolved.toStringExp();
+            TupleExp tup = name ? null : resolved.toTupleExp();
+            if (!tup && !name)
+            {
+                ns.error("expected string expression for namespace name, got `%s`", ns.identExp.toChars());
```

Comment:
The tuple must be strings. I considered this message for a while, but felt that sticking with this message was less confusing. The message is: this expects strings. So explain that it must be a string or a tuple of strings felt too long and potentially more confusing that this message. YMMV?

Reply:
Like, the precision is interesting to a compiler author, but a user just needs to know that they need to supply strings.

Reply:
So maybe use the plural "string expressions"?

Reply:
> To explain that it must be a string or a tuple of strings felt too long and potentially more confusing that this message

In theory it could be an array of strings instead of a tuple, which is not the same thing.